jdaviz/configs/mosviz/plugins/parsers.py

```diff
@@ -283,3 +319,190 @@ def mos_meta_parser(app, data_obj):
+    source_cat = list(p.glob("{}*_direct_*_cat.ecsv".format(obs_label)))
+    direct_image = list(p.glob("{}*_direct_dit1*_cal.fits".format(obs_label)))
+    spec2d_r = list(p.glob("{}*_WFSSR_*_cal.fits".format(obs_label)))
+    spec2d_c = list(p.glob("{}*_WFSSC_*_cal.fits".format(obs_label)))
+    spec1d_r = list(p.glob("{}*_WFSSR_*_x1d.fits".format(obs_label)))
+    spec1d_c = list(p.glob("{}*_WFSSC_*_x1d.fits".format(obs_label)))
+
+    file_lists = {
+                  "Source Catalog": source_cat,
+                  "Direct Image": direct_image,
+                  "2D Spectra C": spec2d_c,
+                  "2D Spectra R": spec2d_r,
+                  "1D Spectra C": spec1d_c,
+                  "1D Spectra R": spec1d_r
+                 }
+
+    # print(len(file_lists["Source Catalog"]), len(file_lists["Direct Image"]), len(file_lists["2D Spectra C"]),
+    #       len(file_lists["2D Spectra R"]), len(file_lists["1D Spectra C"]), len(file_lists["1D Spectra R"]))
+
+    # Convert from pathlib Paths back to strings
+    for key in file_lists:
+        file_lists[key] = [str(x) for x in file_lists[key]]
+
+    found_files = _warn_if_not_found(app, file_lists)
+
+    # Parse relevant information from source catalog
+    cat_fields = ["id", "sky_centroid.ra", "sky_centroid.dec"]
+    source_ids = []
+    ras = []
+    decs = []
+    image_order = []
+
+    # Retrieve source information
+    for source_catalog_num in range(0, len(file_lists["Source Catalog"])):
+        cat_file = file_lists["Source Catalog"][source_catalog_num]
+        parsed_cat_fields = _fields_from_ecsv(cat_file, cat_fields, delimiter=" ")
+
+        for row in parsed_cat_fields:
+            # Only add the first source
+            if row[0] == "1":
+                # This line finds the pupil value in the filename (i.e. F150W)
+                pupil = [x for x in cat_file.split("/")[-1].split("_") if x[0] == "F" or x[0] == "f"][0]
+                image_order.append(pupil)
+                source_ids.append("Source Catalog: {} Source ID: {}".format(pupil, row[0]))
+                ras.append(row[1])
+                decs.append(row[2])
+
+    # Double the source ids, ra, and dec lists to account for C and R orientations
+    source_ids += source_ids
+    ras += ras
+    decs += decs
+
+    _add_to_table(app, source_ids, "Source ID")
+    _add_to_table(app, ras, "Right Ascension")
+    _add_to_table(app, decs, "Declination")
+
+    # Read in direct image filters
+    image_dict = {}
+    filter_wcs = {}
+    for image_file in file_lists["Direct Image"]:
+        im_split = image_file.split("/")[-1].split("_")
+        pupil = [x for x in image_file.split("/")[-1].split("_") if x[0] == "F" or x[0] == "f"][0]
+
+        image_label = "Image {} {}".format(im_split[0], pupil)
+
+        image_data = CCDData.read(image_file)
+        temp = fits.open(image_file)
+        filter_wcs[pupil] = temp[1].header
```

Comment:
Any reason to keep `temp` open?
```suggestion
        with fits.open(image_file) as temp:
            filter_wcs[pupil] = temp[1].header
```

Reply:
Good catch!